sklearn/linear_model/tests/test_sgd.py

```diff
@@ -1338,6 +1387,303 @@ def test_loss_function_epsilon(klass):
+
+
+@pytest.mark.parametrize('klass', [SGDOneClassSVM, SparseSGDOneClassSVM])
+def _test_warm_start_oneclass(klass, X, lr):
+    # Test that explicit warm restart...
+    clf = klass(nu=0.5, eta0=0.01, shuffle=False,
+                learning_rate=lr)
+    clf.fit(X)
+
+    clf2 = klass(nu=0.1, eta0=0.01, shuffle=False,
+                 learning_rate=lr)
+    clf2.fit(X, coef_init=clf.coef_.copy(),
+             offset_init=clf.offset_.copy())
+
+    # ... and implicit warm restart are equivalent.
+    clf3 = klass(nu=0.5, eta0=0.01, shuffle=False,
+                 warm_start=True, learning_rate=lr)
+    clf3.fit(X)
+
+    assert clf3.t_ == clf.t_
+    assert_array_almost_equal(clf3.coef_, clf.coef_)
+
+    clf3.set_params(nu=0.1)
+    clf3.fit(X)
+
+    assert clf3.t_ == clf2.t_
+    assert_array_almost_equal(clf3.coef_, clf2.coef_)
+
+
+@pytest.mark.parametrize('klass', [SGDOneClassSVM, SparseSGDOneClassSVM])
+@pytest.mark.parametrize('lr',
+                         ["constant", "optimal", "invscaling", "adaptive"])
+def test_warm_start_oneclass(klass, lr):
+    _test_warm_start_oneclass(klass, X, lr)
+
+
+@pytest.mark.parametrize('klass', [SGDOneClassSVM, SparseSGDOneClassSVM])
+def test_clone_oneclass(klass):
+    # Test whether clone works ok.
+    clf = klass(nu=0.5)
+    clf = clone(clf)
+    clf.set_params(nu=0.1)
+    clf.fit(X)
+
+    clf2 = klass(nu=0.1)
+    clf2.fit(X)
+
+    assert_array_equal(clf.coef_, clf2.coef_)
+
+
+@ignore_warnings
+@pytest.mark.parametrize('klass', [SGDOneClassSVM, SparseSGDOneClassSVM])
+def test_partial_fit_oneclass(klass):
+    third = X.shape[0] // 3
+    clf = klass(nu=0.1)
+
+    clf.partial_fit(X[:third])
+    assert clf.coef_.shape == (X.shape[1], )
+    assert clf.offset_.shape == (1,)
+    assert clf.predict([[0, 0]]).shape == (1, )
+    id1 = id(clf.coef_.data)
+
+    clf.partial_fit(X[third:])
+    id2 = id(clf.coef_.data)
+    # check that coef_ haven't been re-allocated
+    assert id1 == id2
+
+    # raises ValueError if number of features does not match previous data
+    assert_raises(ValueError, clf.partial_fit, X[:, 1])
+
+
+@pytest.mark.parametrize('klass', [SGDOneClassSVM, SparseSGDOneClassSVM])
+@pytest.mark.parametrize('lr',
+                         ["constant", "optimal", "invscaling", "adaptive"])
+def test_partial_fit_equal_fit_oneclass(klass, lr):
+    clf = klass(nu=0.05, max_iter=2, eta0=0.01,
+                learning_rate=lr, shuffle=False)
+    clf.fit(X)
+    y_scores = clf.decision_function(T)
+    t = clf.t_
+    coef = clf.coef_
+    offset = clf.offset_
+
+    clf = klass(nu=0.05, eta0=0.01, max_iter=1,
+                learning_rate=lr, shuffle=False)
+    for _ in range(2):
+        clf.partial_fit(X)
+    y_scores2 = clf.decision_function(T)
+
+    assert clf.t_ == t
+    assert_array_almost_equal(y_scores, y_scores2)
+    assert_array_almost_equal(clf.coef_, coef)
+    assert_array_almost_equal(clf.offset_, offset)
```

Comment:
I think we are standardizing to `assert_allclose` instead of `assert_array_almost_equal` in scikit-learn nowadays.

For this this does not matter too much, but often it's more natural to pass `atol` or `rtol` or combination of both instead of `decimals`.

Reply:
Similar comment elsewhere in this PR.

Reply:
I used `assert_allclose` everywhere I introduced a new `assert_array_almost_equal`. There are remaining ones (involving other SGD estimators) that should preferably be removed in another PR.